File: ark/api/log.cpp
// Copyright (c) Microsoft Corporation.
// Licensed under the MIT license.

#include "ark/log.hpp"

#include "logging.hpp"

namespace ark {

void log(LogLevel level, const std::string &file, int line,
         const std::string &msg) {
    _log(level, file, line, msg);
}

}  // namespace ark

File: python/ark/log.py
# Copyright (c) Microsoft Corporation.
# Licensed under the MIT license.

import inspect
from .core import LogLevel, log
from .error import *
from .error import __all__ as error_all

__all__ = [*error_all, "DEBUG", "INFO", "WARN"]


def DEBUG(msg: str) -> None:
    frame = inspect.currentframe().f_back
    info = inspect.getframeinfo(frame)
    log(LogLevel.DEBUG, info.filename, info.lineno, msg)


def INFO(msg: str) -> None:
    frame = inspect.currentframe().f_back
    info = inspect.getframeinfo(frame)
    log(LogLevel.INFO, info.filename, info.lineno, msg)


def WARN(msg: str) -> None:
    frame = inspect.currentframe().f_back
    info = inspect.getframeinfo(frame)
    log(LogLevel.WARN, info.filename, info.lineno, msg)
